sources/packages/backend/libs/integrations/src/services/ssh/sftp-integration-base.ts

```diff
@@ -262,13 +267,19 @@ export abstract class SFTPIntegrationBase<DownloadType> {
   }
 
   /**
-   * Delete a file from SFTP.
+   * Archives a file on SFTP to the "Archive\" directory in the original file folder.
    * @param remoteFilePath full remote file path with file name.
    */
-  async deleteFile(remoteFilePath: string): Promise<void> {
+  async archiveFile(remoteFilePath: string): Promise<void> {
     const client = await this.getClient();
     try {
-      await client.delete(remoteFilePath);
+      // Extract the directory from the file path.
+      const directoryPath = path.dirname(remoteFilePath);
+      const fileBaseName = path.basename(remoteFilePath);
+      await client.rename(
```

Comment:
I have a technical question here. The method rename expects that the `toPath` to be renamed should not not already exist and believe it would throw error in that case. 
![image](https://github.com/user-attachments/assets/bba04dba-88c1-4003-8ec4-672108561171)

If that happens due to the reason that same file is sent twice by integration clients, then the file in the source directory will be sitting there for ever and will be processed repeatedly.  
![image](https://github.com/user-attachments/assets/0c6ac529-aeec-4d53-a9a7-0aa513d9ee9f)

Should we pick posixRename() method for this reason ??

Reply:
@andrewsignori-aot @guru-aot @bidyashish @sh16011993 thoughts?

Reply:
@dheepak-aot The idea is not to delete any file. For `posixRename` the doc says "If it already exists, it will be replaced by file specified in fromPath". If that is the case, we should archive renaming the file with a timestamp.

Reply:
We are not certain about what happens when same file comes again as the story does not mention it.

Reply:
I believe that we should not replace the file. If the file was added twice it should be archived twice. Appending a complete timestamp to the file would allow a clear history, IMO.

Reply:
I added a timestamp to the archived file name. Thanks!